api/app/graphql/types/mutation_type.rb

```diff
@@ -3,10 +3,28 @@
 module Types
   class MutationType < Types::BaseObject
     # TODO: remove me
-    field :test_field, String, null: false,
-                               description: 'An example field added by the generator'
-    def test_field
-      'Hello World'
+    field :register, Types::UserType, null: false do
+      argument :phone, String, required: true
+      argument  :email, String, required:false
+      argument :date_of_birth, GraphQL::Types::ISO8601DateTime, required: true
+      argument :isMuslim, Boolean, required: false
+      argument :password, String, required: true
+      argument :password_confirmation, String, required: true
+
+    end
+  
+    def register(**kwargs)
+    
+      user = User.create!(
+        kwargs
+      )
+
+      user 
```

Comment:
Instead of returning the entire user object, you can return an object that contains only the following:

- [ ] `authToken` - The JWT token created after creating an account if successful.
- [ ] `status` - Whether the request was successful or not.
- [ ] `message` - If execution fails then indicate reason for failure using this property.